### PR TITLE
RLP encoding and decoding

### PR DESCRIPTION
**Pull request by Sarvesh19**
Added encoding and decoding part but test cases is remaining

### CODE REVIEW

**Comment by akshatdev0** [comment]
@Sarvesh19 
Please try to make commits as short as and as frequent as possible. How come you can commit 4 files and 400 lines of code in a single commit. See my commit history on my branch **eckey**

**Comment by Sarvesh19** [comment]
@akshatdev0  I will try to make as short as possible from next time

**Comment by bhaskar6473** [comment]
@Sarvesh19  , please make commit like one complete change in one commit, for example "encoding for for numbers " in one commit, and "encoding for strings " in one , complete change

**Review by bhaskar6473 — APPROVED** [review]

File: src/test/java/studio/blockops/vyom/core/utils/RLPTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package studio.blockops.vyom.core.utils;
+
+
+
+import org.bouncycastle.util.encoders.Hex;
+// Please delete this class after unit test class
+public class RLPTest {
+	public static void main(String[] args) {
```
Comment (akshatdev0) [inline src/test/java/studio/blockops/vyom/core/utils/RLPTest.java]: Follow proper unit test flow, as performed in eckey.
Unit tests are not performed using main() method.
No System logs in unit tests.
Tests will be in src/test/java source package.

File: src/main/java/studio/blockops/vyom/core/utils/Value.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+//
+//import java.math.BigInteger;
+//
+//import java.util.Arrays;
+//import java.util.List;
+
+/**
+ * Class to encapsulate an object and provide utilities for conversion
+ */
+public class Value {
```
Comment (akshatdev0) [inline src/main/java/studio/blockops/vyom/core/utils/Value.java]: Provide proper name to the class. Current name doesn't reflect it's purpose.

File: src/test/java/studio/blockops/vyom/core/utils/RLPTest.java (hunk 1 above)
Comment (Sarvesh19) [inline src/test/java/studio/blockops/vyom/core/utils/RLPTest.java:8]: I just mentioned @akshatdev0  // Please delete this class after unit test class
Need to remove this i know No System logs in unit tests.

File: src/main/java/studio/blockops/vyom/core/utils/Value.java (hunk 2 above)
Comment (Sarvesh19) [inline src/main/java/studio/blockops/vyom/core/utils/Value.java:17]: I think it does

File: src/main/java/studio/blockops/vyom/core/utils/Value.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+	private Object value;
```
Comment (akshatdev0) [inline src/main/java/studio/blockops/vyom/core/utils/Value.java:19]: Enforce immutability wherever you can and wherever possible.

File: src/main/java/studio/blockops/vyom/core/utils/RLP.java [hunk 4]
```diff
@@ -0,0 +1,272 @@
+package studio.blockops.vyom.core.utils;
```
Comment (akshatdev0) [inline src/main/java/studio/blockops/vyom/core/utils/RLP.java:1]: Try to look at **org.nem.core.serialization** package and see how they have implemented their serialization.
They have better implementation of encoding/decoding. Can use their pattern and ethereum RLP encoding technique.

File: src/main/java/studio/blockops/vyom/core/utils/Value.java (hunk 2 above)
Comment (akshatdev0) [inline src/main/java/studio/blockops/vyom/core/utils/Value.java:17]: Provide proper name to the class. Current name doesn't reflect it's purpose.

File: src/test/java/studio/blockops/vyom/core/utils/RLPTest.java (hunk 1 above)
Comment (akshatdev0) [inline src/test/java/studio/blockops/vyom/core/utils/RLPTest.java:8]: Follow proper unit test flow, as performed in eckey.
Unit tests are not performed using main() method.
No System logs in unit tests.
Tests will be in src/test/java source package.

**Review by akshatdev0 — CHANGES_REQUESTED** [review]

**Comment by akshatdev0** [comment]
We will start fresh for RLP with proper design.

**Pull request closed without merging**